### DISCUSSION TITLE
Is it still not possible to restrict by ip?

### DISCUSSION THREAD

**Discussion by Zerebokep** [vercel/community#1010] [Help] [answered]
Title. I would like to block access to a deployment except for one IP(-range).

**Comment by amyegan** [comment, marked as answer]
Hi @Zerebokep! Password or SSO protection are the recommended ways to [restrict deployment access](https://vercel.com/blog/protecting-deployments). Or you can use [Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) to block/allow IPs programmatically.
